### PR TITLE
- Add retry feature test

### PR DESCRIPTION
**Pull request by amamounelsayed**
This PR is providing the e2e test for the retry feature.

### CODE REVIEW

File: setup-tests-pipeline.ps1 [hunk 1]
```diff
@@ -17,13 +17,18 @@ if(!$skipCliDownload)
...
 
-  if ($version -and $version.trim())
+  if (-not($version -and $version.trim()))
   {
     $env:CORE_TOOLS_URL = "https://pgopafunctestv2storage.blob.core.windows.net/cli/pgopa-cli.zip"
+    Write-Host "Using Functions Core Tools version: pgopa-CLI"
+  }
+  else
+  {
+    $env:CORE_TOOLS_URL = "https://functionsclibuilds.blob.core.windows.net/builds/3/latest/Azure.Functions.Cli.win-x64.zip"
```
Comment (TsuyoshiUshio) [inline setup-tests-pipeline.ps1:30]: You are changing the core tools from official blob? Looks great.

File: endtoendtests/Azure.Functions.Java.Tests.E2E/Azure.Functions.Java.Tests.E2E/Constants.cs [hunk 2]
```diff
@@ -20,9 +20,9 @@ public static class Constants
         public static string TestQueueMessage = "Hello, World";
 
         //Blob tests
-        public static string TriggerInputBindingBlobContainer = "test-triggerinput-java";
-        public static string InputBindingBlobContainer = "test-input-java";
-        public static string OutputBindingBlobContainer = "test-output-java";
+        public static string TriggerInputBindingBlobContainer = "test-triggerinput-java-new";
```
Comment (TsuyoshiUshio) [inline endtoendtests/Azure.Functions.Java.Tests.E2E/Azure.Functions.Java.Tests.E2E/Constants.cs:23]: We can create an issue to fix this. (I mean update the pipeline or clean up code.)

**Review by TsuyoshiUshio — APPROVED** [review]
I'd like to ask to create an issue to track the CI issue of clean up. otherwise, LGTM.

File: endtoendtests/Azure.Functions.Java.Tests.E2E/Azure.Functions.Java.Tests.E2E/Constants.cs (hunk 2 above)
Comment (amamounelsayed) [reply endtoendtests/Azure.Functions.Java.Tests.E2E/Azure.Functions.Java.Tests.E2E/Constants.cs:23]: Yes great point looks like the createBlob and cleanBlob for a reason not working in the e2e test.

File: setup-tests-pipeline.ps1 (hunk 1 above)
Comment (amamounelsayed) [reply setup-tests-pipeline.ps1:30]: Correct, We are getting the latest release version

**Review by TsuyoshiUshio — APPROVED** [review]
LGTM

**Pull request merged**